.github/workflows/validate.yml

```
@@ -0,0 +1,49 @@
+name: validate
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - master
+      - main
+  pull_request:
+    branches:
+      - master
+      - main
+
+jobs:
+  linters:
+    permissions:
+      contents: read      # for actions/checkout to fetch code
+      pull-requests: read # for golangci/golangci-lint-action to fetch pull requests
+    timeout-minutes: 10
+
+    strategy:
+      matrix:
+        # We only run on the latest version of go, as some linters may be
+        # version-dependent (for example gofmt can change between releases).
+        go-version: [1.19.x]
```

```suggestion
        go-version: ['stable']
```

Again, I realize it's non-deterministic, but I think for this particular repo it's probably the better way to go...